projects/assets-library/src/icons/icon-registry.service.ts

```diff
@@ -1,16 +1,23 @@
 import { Injectable } from '@angular/core';
 import { MatIconRegistry } from '@angular/material/icon';
 import { DomSanitizer } from '@angular/platform-browser';
-import { isEmpty } from 'lodash-es';
+import { isEmpty, orderBy } from 'lodash-es';
 import { Observable } from 'rxjs';
 
 @Injectable({ providedIn: 'root' })
 export class IconRegistryService {
   private static readonly SVG_ICON_PREFIX: string = 'svg:';
 
+  private registeredIcons: string[] = [];
+
   public constructor(private readonly matIconRegistry: MatIconRegistry, private readonly sanitizer: DomSanitizer) {}
 
   public registerSvgIcon(iconInfo: SvgIconRegistrationInfo): void {
+    if (!this.registeredIcons.includes(iconInfo.key)) {
```

Comment:
Qq: can we pull this from matIconRegistry?

Reply:
@itssharmasandeep Also, this is taking O(n2). Why not use a map?

Reply:
Right, used set instead, here is the follow up PR https://github.com/hypertrace/hypertrace-ui/pull/2304#discussion_r1261592929